packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx

```diff
@@ -60,18 +70,33 @@ export const PlaylistsTab = () => {
             (trackIds.length === 0 ||
               trackIds.some((t) => {
                 return (
+                  offlineDownloadStatus &&
                   offlineDownloadStatus[t.toString()] ===
-                  OfflineDownloadStatus.SUCCESS
+                    OfflineDownloadStatus.SUCCESS
                 )
               }))
           )
         }
         return true
       })
     },
-    [filterValue, isReachable, isOfflineModeEnabled, isDoneLoadingFromDisk]
+    [
+      filterValue,
+      isReachable,
+      isOfflineModeEnabled,
+      offlineDownloadStatus,
+      isDoneLoadingFromDisk
+    ]
   )
 
+  const numPlayableTracksMap = useMemo(() => {
+    return buildCollectionIdsToNumPlayableTracksMap(
```

Comment:
This solution feels very jank but it's the best I could come up with given how the code is structured, and while keeping CollectionList and its children purely presentational. Very open to other solutions if this is too cumbersome!

Reply:
yeah i think this feeling is potentially resolved with the ideas below?